### ISSUE TITLE
empty array of record_ids issues a DELETE SQL statement

### ISSUE THREAD

**Issue by rjnienaber** [jaynetics/delete_recursively#3]
Here's a contrived example of what I mean:
```
 require 'bundler/inline'
 
 gemfile do
   source 'https://rubygems.org'
   gem 'activerecord', '~> 7.0', '>= 7.0.4.3'
   gem 'delete_recursively'
   gem 'sqlite3', '~> 1.6', '>= 1.6.2'
 end
 
 ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')
 
 class Task < ActiveRecord::Base
   has_one :status, dependent: :destroy
 end
 
 class Status < ActiveRecord::Base
   belongs_to :status
 end
 
 ActiveRecord::Schema.define do
   create_table(:tasks) { |t| t.string :description }
   create_table(:statuses) { |t| t.references :task }
 end
 
 task = Task.create(description: "Buy fish and chips")
 
 ActiveRecord::Base.logger = ActiveSupport::Logger.new($stdout)
 
 DeleteRecursively.delete_records_recursively(Task, task.id, force: true)
```
This generates the following output:
```
 -- create_table(:tasks)
    -> 0.0093s
 -- create_table(:statuses)
    -> 0.0002s
   Status Pluck (0.0ms)  SELECT "statuses"."id" FROM "statuses" WHERE "statuses"."task_id" = ?  [["task_id", 1]]
   Status Delete All (0.0ms)  DELETE FROM "statuses" WHERE 1=0
   Task Delete All (0.0ms)  DELETE FROM "tasks" WHERE "tasks"."id" = ?  [["id", 1]]
```
The `Status Delete All` statement above seems unnecessary and is being issued because ActiveRecord is being asked to delete rows with an empty array of ids.

Is this intended? Perhaps it makes sense to add an `if record_ids.size > 0` when:
* [recursing](https://github.com/jaynetics/delete_recursively/blob/v1.2.1/lib/delete_recursively.rb#L36)
* [deleting](https://github.com/jaynetics/delete_recursively/blob/v1.2.1/lib/delete_recursively.rb#L43)

**Comment by jaynetics** [comment]
excellent catch! you would do well in QA 😄 

a recursion guard was already [added in v1.2.1](https://github.com/jaynetics/delete_recursively/blob/e76628462a5495c1574d1583cf46943cf6f4831e/lib/delete_recursively.rb#L20).

regarding the delete calls, i had no idea that activerecord calls the DB with empty id lists.

it does make sense that they do it, though, from a consistency and modularity point of view.

i've just released v1.2.2 with the proposed change. thanks for the report!

**Comment by rjnienaber** [comment]
Awesome. Thanks for being so responsive 🙂.
